### PR TITLE
palomar: refactor query string parsing; updates to schemas and transform

### PR DESCRIPTION
**Pull request by bnewbold**
TODO:

- [x] normalize URLs: maybe with `purell`, maybe stripping tracking parameters
- [x] date-level queries
- [x] add end-to-end tests (similar to japanese queries)
- [x] implement "merge" of query params (parsed and actual params)
- [x] run lexgen and wire up actual query params
- [x] `from:me` post search behavior (server-side, using viewer param)
- [ ] alternate post "sort" order (top vs. latest)

Closes: https://github.com/bluesky-social/indigo/issues/191
Closes: https://github.com/bluesky-social/indigo/issues/593

### CODE REVIEW

File: search/url.go [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+// aggressively normalizes URL, for search indexing and matching. it is possible the URL won't be directly functional after this normalization
+func NormalizeLossyURL(raw string) string {
+	clean, err := purell.NormalizeURLString(raw, purell.FlagsUsuallySafeGreedy|purell.FlagRemoveDirectoryIndex|purell.FlagRemoveFragment|purell.FlagRemoveDuplicateSlashes|purell.FlagRemoveWWW|purell.FlagSortQuery)
+	if err != nil {
+		return raw
+	}
+
+	// remove tracking params
+	u, err := url.Parse(clean)
```
Comment (ericvolp12) [inline search/url.go:44]: Think we need some error handling here. `u` can be nil and we ignore this error

**Review by ericvolp12 — APPROVED** [review]
yolo

File: search/handlers.go [hunk 2]
```diff
@@ -76,16 +76,132 @@ func (s *Server) handleSearchPostsSkeleton(e echo.Context) error {
...
+				"message": fmt.Sprintf("invalid Language for 'lang': %s", err),
+			})
+		}
+		params.Lang = &l
+	}
+	// TODO: could be multiple tag params; guess we should "bind"?
+	tagStr := e.QueryParam("tag")
+	if tagStr != "" {
+		params.Tags = []string{tagStr}
+	}
```
[the remark this answers is not in the document]
Comment (bnewbold) [reply search/handlers.go:191]: thanks!

**Comment by bnewbold** [comment]
going to do the query DSL tweaks (for "top", etc) in a follow-up!

**Pull request merged**